doc/api/assert.markdown

```diff
@@ -2,42 +2,131 @@
 
     Stability: 3 - Locked
 
-This module is used so that Node.js can test itself. It can be accessed with
-`require('assert')`. However, it is recommended that a userland assertion
-library be used instead.
+The `assert` module was originally written and included in Node.js in order
+to facilitate Node.js' own unit tests. While it can be used by JavaScript
+running within Node.js by using `require('assert')`, it is recommended that a
```

Comment:
Although now I'm trying to find that conversation/discussion and can't so maybe I imagined it.